BigEval.js

```diff
@@ -570,6 +570,12 @@ BigEval.prototype.compile = function (expression) {
 			((i > 1 && tokens[i - 2].type === TokenType.OP) || i === 1)
 			) {
 
+			console.log(tokens[i - 2]);
+
+			if (i > 1 && tokens[i - 2].value==='!') {				
```

Comment:
If i = 2, wouldn't `!` being at token 0 be an invalid case?

I haven't looked at the code in years, so I apologize if I am wrong.

Also, can you please remove the console.log? Thanks.

Reply:
Oops. Sorry. That should be sorted now.

Reply:
Re I=2 trouble is ! can also be used as Not operator .. ?

Reply:
You are right. I forgot we had NOT operator.